networkx/algorithms/sampling.py

```diff
@@ -0,0 +1,278 @@
+# -*- encoding: utf-8 -*-
+# sampling.py - functions for sampling nodes from a graph
+#
+# Copyright 2011 Maciej Kurant.
+# Copyright 2015 NetworkX developers.
+#
+# This file is part of NetworkX.
+#
+# NetworkX is distributed under a BSD license; see LICENSE.txt for more
+# information.
+"""Provides functions for sampling nodes from a graph."""
+from __future__ import division
+
+import bisect
+import itertools
+import random
+
+import networkx as nx
+
+__all__ = ['degree_weighted_independent_node_sample', 'random_walk',
+           'uniform_independent_node_sample']
+
+
+# def estimate_mean(sample, values, weights=None):
```

Comment:
I think it is questionable whether any of these functions belong in the
public networkx API.


On Thu, May 21, 2015 at 9:52 AM, jfinkels <notifications@github.com> wrote:

> In networkx/algorithms/sampling.py
> <https://github.com/networkx/networkx/pull/1536#discussion_r30816423>:
>
> > +# NetworkX is distributed under a BSD license; see LICENSE.txt for more
> > +# information.
> > +"""Provides functions for sampling nodes from a graph."""
> > +from __future__ import division
> > +
> > +import bisect
> > +import itertools
> > +import random
> > +
> > +import networkx as nx
> > +
> > +__all__ = ['degree_weighted_independent_node_sample', 'random_walk',
> > +           'uniform_independent_node_sample']
> > +
> > +
> > +# def estimate_mean(sample, values, weights=None):
>
> I question whether this should function should be provided by NetworkX.
> This function has nothing in particular to do with graphs, so it should not
> be provided as part of the public API of NetworkX. At best, these should be
> provided as examples only.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/networkx/networkx/pull/1536/files#r30816423>.
>